src/functional/program_processors/codegens/gtfn/itir_to_gtfn_ir.py

```diff
@@ -69,10 +69,10 @@ def pytype_to_cpptype(t: str):
 _horizontal_dimension = "gtfn::unstructured::dim::horizontal"
 
 
-def _get_domains(closures: Iterable[itir.StencilClosure]) -> Iterable[itir.FunCall]:
+def _get_domains(closures: Iterable[itir.StencilClosure]) -> list[itir.FunCall]:
```

Comment:
replace by just
```
def _get_domains(closures: Iterable[itir.StencilClosure]) -> Iterable[itir.FunCall]:
    return (c.domain for c in closures)
```
and fix `itir.StencilClosure` `domain` to be `itir.FunCall` in `ir.py`. It seems we heavily rely on that...